Lib/test/test_struct.py

```diff
@@ -935,6 +935,14 @@ def test_half_float(self):
             self.assertTrue(math.isnan(struct.unpack('<e', bits)[0]))
             self.assertTrue(math.isnan(struct.unpack('>e', bits[::-1])[0]))
 
+        # Check round-trip for NaN's:
+        if sys.platform != 'win32':
```

Comment:
Why is the tet skipped on Windows?

Reply:
I tried to enable new test on win32:
https://github.com/python/cpython/actions/runs/13514756293/job/37761342232
and it looks like the round-trip works here only for qNaN's.  sNaN's stored as qNaN's even with PyFloat_Unpack8().

The original IEEE 754 didn't specify how the NaN type should be tagged, it's coming with 2008 revision.